File: LB-Assignments/Assignment_35/program3.cpp
/*
accept number from user and check whether
7th, 15th, 21st and 28th bit os on or off
*/
#include<iostream>
using namespace std;

typedef unsigned int UINT;

bool ChkBits(UINT iNo)
{
    UINT iAns = 0x0;
    bool bResult = false;
    UINT iMask = 0x08104040;

    iAns = iNo & iMask;

    if(iAns == iMask)
    {
        bResult = true;
    }

    return bResult;

}
int main()
{
    UINT iValue = 0;
    bool bRet = false;

    cout<<"Enter number : \n";
    cin>>iValue;

    bRet = ChkBits(iValue);

    if (bRet)
    {
        cout<<"Bits are on\n";
    }
    else
    {
        cout<<"Bits are off\n";
    }
    
    return 0;
}

File: LB-Assignments/Assignment_38/program3.java
/*
Accept 1 number from user 
and check whether 9th or 12 th bit is on or off
*/
import java.util.*;

class Bitwise
{
    public int iNo;

    public Bitwise(int a)
    {
        this.iNo = a;
    }

    public boolean ChkBits()
    {
        boolean bRet = false;

        int iMask = 0X900;
        int Ans = iNo & iMask;

        if((Ans == 0x800) || (Ans == 0x100))
        {
            bRet = true;
        }
        return bRet;

    }
}
class program3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue= 0;

        System.out.println("Enter First Number : ");
        iValue = sobj.nextInt();

        Bitwise bobj = new Bitwise(iValue);

        if(bobj.ChkBits())
        {
            System.out.println("TRRUE");
        }
        else
        {
            System.out.println("FALSE");
        }

    }
}